ssllib.c

```diff
@@ -307,6 +307,23 @@ ssl_strdup(const char *s)
     return new;
 }
 
+static RSA *
+ssl_rsadup(const RSA *rsa)
+{
+    RSA *c = RSA_new();
+
+    c->n = rsa->n;
```

Comment:
Why not simply *c = *rsa?  That seems more robust to me.   You probably also have to check
that `c` is not `NULL`.